### PR TITLE
#5533 Enable in-marketplace one-click activation for mods with built-in service configs

### LINKED ISSUE #5533: Stop displaying Integration selection at Marketplace mod activation (or reactivation) when there is a built-in integration and no other configured integration
Slice 1: <https://www.notion.so/pixiebrix/Remove-the-need-to-configure-certain-integrations-ceb6d844c3504933af398925448bb6fb>

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Closes #5533
- Refactors `canAutoActivate` into a `useCanAutoActivate` hook
- Introduces a `useBuiltInServiceAuthsByRequiredServiceIds` hook
- Adds tests for auto-activating mods with and without built-in service configurations

## Reviewer Tips

- Pre-fills services with built-in configurations in `useWizard` (prefers configurations that are already installed)
- Modifies `canAutoActivate` to be truthy if all services have built-in configurations
- Moves `canAutoActivate` to `RequireRecipe` component to reduce complexity around component flickering while fetching built-in auths

## Demo

- https://www.loom.com/share/4df37c03df1846309738f9f452f7cc1c

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer @BLoe

### CODE REVIEW

File: src/activation/useWizard.ts [hunk 1]
```diff
@@ -100,7 +103,7 @@ function useWizard(
       services: serviceIds.map((id) => ({
         id,
         // eslint-disable-next-line security/detect-object-injection -- is a registry id
-        config: installedServices[id],
+        config: installedServices[id] ?? builtInServiceAuths[id],
```
Comment (mnholtz) [inline src/activation/useWizard.ts:106]: Prefer installed service configs to built-in service configs

File: src/background/starterBlueprints.test.ts [hunk 2]
```diff
@@ -74,46 +74,6 @@ beforeEach(async () => {
...
-            name: "Service 2",
-          },
-        },
-      }),
-    }),
-  ];
-
-  return { recipe, builtInServiceAuths };
-};
-
```
Comment (mnholtz) [inline src/background/starterBlueprints.test.ts:116]: Moved to factories.ts

File: src/sidebar/activateRecipe/ActivateRecipePanel.tsx [hunk 3]
```diff
@@ -98,25 +96,11 @@ const activationSlice = createSlice({
...
-    )
-  );
-  const needsServiceInputs = recipeServiceIds.some(
-    (serviceId) => serviceId !== PIXIEBRIX_SERVICE_ID
-  );
-
-  // Can auto-activate if no configuration required
-  return !hasRecipeOptions && !needsServiceInputs;
-}
-
```
Comment (mnholtz) [inline src/sidebar/activateRecipe/ActivateRecipePanel.tsx:115]: Changed to a hook & moved to `RequireRecipe`

File: src/sidebar/activateRecipe/RequireRecipe.tsx [hunk 4]
```diff
@@ -25,18 +25,56 @@ import styles from "./RequireRecipe.module.scss";
...
   includesQuickBar: boolean;
+  canAutoActivate: boolean;
 };
 
 type RequireRecipeProps = {
   recipeId: RegistryId;
   children: (props: RecipeState) => React.ReactElement;
 };
 
+function useCanAutoActivate(recipe: RecipeDefinition | null): {
```
[the remark this answers is not in the document]
Comment (mnholtz) [reply src/sidebar/activateRecipe/RequireRecipe.tsx:44]: Realized that this as a bullet included in the spec. I missed this. Will add.

File: src/components/fields/schemaFields/widgets/ServiceWidget.test.tsx
Comment (mnholtz) [inline src/components/fields/schemaFields/widgets/ServiceWidget.test.tsx:1]: Refactoring changes

File: src/components/fields/schemaFields/widgets/ServiceWidget.tsx
Comment (mnholtz) [inline src/components/fields/schemaFields/widgets/ServiceWidget.tsx:1]: `useAuthOptions` refactoring changes

File: src/contrib/automationanywhere/BotOptions.test.tsx
Comment (mnholtz) [inline src/contrib/automationanywhere/BotOptions.test.tsx:1]: `useAuthOptions` refactoring changes

File: src/contrib/uipath/LocalProcessOptions.test.tsx
Comment (mnholtz) [inline src/contrib/uipath/LocalProcessOptions.test.tsx:1]: `useAuthOptions` refactoring changes

File: src/contrib/uipath/ProcessOptions.test.tsx
Comment (mnholtz) [inline src/contrib/uipath/ProcessOptions.test.tsx:1]: `useAuthOptions` refactoring changes

File: src/extensionConsole/pages/activateExtension/ActivatePage.tsx
Comment (mnholtz) [inline src/extensionConsole/pages/activateExtension/ActivatePage.tsx:1]: `useAuthOptions` refactoring changes

File: src/extensionConsole/pages/activateRecipe/ActivateWizard.test.tsx
Comment (mnholtz) [inline src/extensionConsole/pages/activateRecipe/ActivateWizard.test.tsx:1]: `useAuthOptions` refactoring changes

File: src/extensionConsole/pages/activateRecipe/ServicesBody.tsx
Comment (mnholtz) [inline src/extensionConsole/pages/activateRecipe/ServicesBody.tsx:1]: `useAuthOptions` refactoring changes

**Review by BLoe — APPROVED** [review]
LGTM 🚀

**Pull request merged**